File: test/scope/valid/valid_scope.go
package main 

import "fmt" 

var b int = 2
func f(c int)   {
	fmt.Printf("FUNCTION: local var:%d  \t global var: %d \n", c, b)
	for b := 0; b<2; b=b+1 {
		fmt.Printf("For scope %d \t func variable %d \n", b, c)
	}
}

func main()   {
	var a int = 1
	fmt.Printf("MAIN: local var:%d  \t global var: %d \n", a, b)
	f(3)
}

File: test/scope/valid/valid_scope.c
int b = 2;
void f(int c){
    printf("FUNCTION: local var:%d  \t global var: %d \n", c, b);
    for(int b=0; b<2; b = b+ 1){
        printf("For scope %d \t func variable %d \n", b, c);
    }
}

void main(){
    int a = 1;
    printf("MAIN: local var:%d  \t global var: %d \n", a, b);
    f(3);
}